lib/fulfillment_resolver.js

```diff
@@ -0,0 +1,56 @@
+const { isInRecap } = require('./util')
+
+class FulfillmentResolver {
+  constructor (responseReceived) {
+    this.elasticSearchResponse = responseReceived
+  }
+  responseWithFulfillment () {
+    for (let hit of this.elasticSearchResponse.hits.hits) {
+      (hit._source.items || []).map((item) => {
+        if (item.electronicLocator) return item
+        const _isInRecap = isInRecap(item)
+        item.physFulfillment = this._determinePhysFulfillment(item, _isInRecap, false)
+        item.eddFulfillment = this._determineEddFulfillment(item, _isInRecap, true)
+        return item
+      })
+    }
+    return this.elasticSearchResponse
+  }
+
+  _onsiteLocation (onsiteLocationId) {
+    if (onsiteLocationId === 'sc') return 'sc'
+    else if (onsiteLocationId === 'ma') return 'sasb'
+    else if (onsiteLocationId === 'my') return 'lpa'
+  }
+
+  _recapLocation (recapDepository) {
+    if (recapDepository !== 'hd') return 'recap'
+    else return 'hd'
+  }
+
+  _fulfillmentPrefix (item, _isInRecap) {
+    return _isInRecap
+      ? this._recapLocation(item.recapDepository)
+      : this._onsiteLocation(item.holdingLocation && item.holdingLocation.id)
```

Comment:
Seems like `_onsiteLocation` is written to expect a single building level location id without any prefix (e.g. "ma", "sc"), but what you're sending in is `item.holdingLocation.id`. But `item.holdingLocation.id` will always be `undefined` because `item.holdingLocation` is an array. So fix that. But also, `item.holdingLocation[0].id` will be a prefixed holding location id (e.g. `loc:mal92`, `loc:rcnpa`), so you will also need to 1) drop the prefix and 2) determine the building level location id (e.g. for "mal" the parent is "ma")

Reply:
fixed